comments_blackbox_test.go

```diff
@@ -0,0 +1,201 @@
+package main_test
+
+import (
+	"crypto/rsa"
+	"fmt"
+	"testing"
+
+	"golang.org/x/net/context"
+
+	. "github.com/almighty/almighty-core"
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/app/test"
+	"github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/gormapplication"
+	"github.com/almighty/almighty-core/gormsupport/cleaner"
+	"github.com/almighty/almighty-core/migration"
+	"github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/resource"
+	testsupport "github.com/almighty/almighty-core/test"
+	almtoken "github.com/almighty/almighty-core/token"
+	"github.com/almighty/almighty-core/workitem"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// a normal test function that will kick off TestSuiteComments
+func TestSuiteComments(t *testing.T) {
+	resource.Require(t, resource.Database)
+	suite.Run(t, new(CommentsSuite))
+}
+
+// ========== TestSuiteComments struct that implements SetupSuite, TearDownSuite, SetupTest, TearDownTest ==========
+type CommentsSuite struct {
+	suite.Suite
+	workitemCommentsCtrl app.WorkItemCommentsController
+	workitemCtrl         app.WorkitemController
+	defaultSvc           *goa.Service
+	userSvc              *goa.Service
+	userSvc2             *goa.Service
+	db                   *gorm.DB
+	pubKey               *rsa.PublicKey
+	priKey               *rsa.PrivateKey
+	clean                func()
+	commentId            uuid.UUID
+}
+
+func (s *CommentsSuite) SetupSuite() {
+	var err error
+	if err = configuration.Setup(""); err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+	s.db, err = gorm.Open("postgres", configuration.GetPostgresConfigString())
+	if err != nil {
+		panic("Failed to connect database: " + err.Error())
+	}
+	// default service (without auth)
+	s.defaultSvc = goa.New("Comments-service-test")
+	// user service (with auth)
+	s.pubKey, _ = almtoken.ParsePublicKey([]byte(almtoken.RSAPublicKey))
+	s.priKey, _ = almtoken.ParsePrivateKey([]byte(almtoken.RSAPrivateKey))
+	s.userSvc = testsupport.ServiceAsUser("CommentsSuite-Service", almtoken.NewManager(s.pubKey, s.priKey), account.TestIdentity)
+	s.userSvc2 = testsupport.ServiceAsUser("CommentsSuite-Service", almtoken.NewManager(s.pubKey, s.priKey), account.TestIdentity2)
```

Comment:
Could you split this up into functions/methods instead to make it a bit clearer what the Test case requires? e.g. https://github.com/almighty/almighty-core/blob/master/space_test.go#L43